tracer/src/Datadog.Trace/Debugger/Expressions/ProbeExpressionParser.Collection.cs

```diff
@@ -171,16 +172,16 @@ private MethodCallExpression CollectionCountExpression(Expression source)
         return Expression.Call(source, countOrLength);
     }
 
-    private bool IsSafeCollection(Expression source)
+    private bool IsSafeCollection(Type type)
```

Comment:
I'm confused, I thought we only consider a specific hard-coded list of collections as "safe"? This implementation means any class that implements `ICollection` would be considered safe?

Reply:
The thought behind that was that any object that has an indexer could be used to dump his items